server/src/main/java/com/ncc/neon/controllers/BetterController.java

```diff
@@ -129,7 +136,29 @@
         return res;
     }
 
-    @GetMapping(path="preprocess")
+    @GetMapping(path="irsearch")
+    Mono<Object> irsearch(@RequestParam("query") String query, @RequestParam("module") String module)  {
+        // synchronous service
+        return moduleService.buildNlpModuleClient(module)
+                .flatMap(nlpModule -> {
+                    IRNlpModule irModule = (IRNlpModule) nlpModule;
+                    return irModule.searchIR(query);
+                });
+        // static cast from nlpModule -> IR
+        // return Mono String[]
+    }
+
+    @GetMapping(path = "getdoc")
+    Mono<?> getdoc(@RequestParam("docId") String id) {
```

Comment:
Why not return the actual type?

Reply:
I changed this to a mono<BetterFile> but I'm not sure if we're actually using this endpoint since the frontend doesn't call this endpoint. The function getById inside the elasticsearchService which is what this function getdoc is calling was last updated from January of this year. I'm not sure if we intended to use this endpoint but we abandoned once the scope of the ir document changed but I'll leave it for now.

Reply:
Darian clarified that this was an out of date endpoint that we should delete so I deleted it.